### ISSUE TITLE
Update Middleware Instrumentation for Rails 7.0

### ISSUE THREAD

**Issue by kaylareopelle** [newrelic/newrelic-ruby-agent#881]
# Overview
When running the unit test suite against Rails version 7.0.0.rc1 on Ruby 2.7.x and Ruby 3.0.x, the following errors are raised within the `MiddlewareInstrumentationTest`: 

* **Error:** ```MiddlewareInstrumentationTest#test_rails_middlewares_constructed_by_name: NamedMiddleware should have been called, but wasn't```
* **Test Line:** `test/multiverse/suites/rails/middleware_instrumentation_test.rb:27`

* **Error:** ```MiddlewareInstrumentationTest#test_rails_middlewares_passed_as_instances: InstanceMiddleware should have been called, but wasn't```
* **Test Line:** `test/multiverse/suites/rails/middleware_instrumentation_test.rb:33`

### Prerequisites
* Have Ruby 2.7.5 and 3.0.3 installed (for help installing Ruby versions, we often use [rbenv](https://github.com/rbenv/ruby-build))
* Create a branch off of `test/rails-7-upgrade`

To run the failed file on Rails 7.0: `bundle exec rake "test:multiverse[rails,env=0,file=test/multiverse/suites/rails/middleware_instrumentation_test.rb]"` 

See the [Multiverse README](https://github.com/newrelic/newrelic-ruby-agent/blob/dev/test/multiverse/README.md) for more ways to run the test suite.

**Comment by kaylareopelle** [comment]
It isn't immediately clear to us what's wrong here. It's a ballpark estimate between S/M depending on how we triage the problem. We're not sure if we need to change how Middleware is instrumented.

**Comment by tenpaiyomi** [comment]
To add to this, @kaylareopelle, I would recommend probably updating the priority pretty high.  I updated my company's application to Rails 7.0 (also on Ruby 3.0, but this was prior to the Rails 7.0 update) and began to get an `ActionController::InvalidAuthenticityToken` with error message of `Can't verify CSRF token authenticity.`

After spending a considerable amount of time digging into the various stack traces and all without much success, I began disabling gems which modified the request middleware.  It was only after disabling NewRelic that all `InvalidAuthenticityToken` errors disappeared.  Upon re-enabling the NewRelic gem, the problem immediately returned.

If there is any additional information I can provide, feel free to let me know and I'll provide whatever I can.

**Comment by tannalynn** [comment]
@tenpaiyomi 
Thank you for letting us know about this error. We will make sure we look into that as we're going through the changes we'll need to make to add support rails 7. I'll leave this  as part of this existing ticket for now, but if we find the error seems to be unrelated to these specific test failures we will move this to a separate ticket to track it better.
Thank you for letting us know!

**Comment by kaylareopelle** [comment]
Hello @tenpaiyomi! After looking into your issue, it seems unrelated to this bug. Could you fill in the remainder of the description on issue #887 with additional information about your application?

**Comment by kaylareopelle** [comment]
Closed by merging #886 into the rails 7.0 test branch
